pinot/generative/torch_gvae/utils.py

```diff
@@ -278,4 +141,49 @@ def ismember(a, b, tol=5):
 
     # NOTE: these edge lists only contain single direction of edge!
     return adj_train, train_edges, val_edges, val_edges_false, test_edges, test_edges_false
- 
\ No newline at end of file
+
+
+def preprocess_graph(adj):
+    adj = sp.coo_matrix(adj)
+    adj_ = adj + sp.eye(adj.shape[0])
+    rowsum = np.array(adj_.sum(1))
+    degree_mat_inv_sqrt = sp.diags(np.power(rowsum, -0.5).flatten())
+    adj_normalized = adj_.dot(degree_mat_inv_sqrt).transpose().dot(degree_mat_inv_sqrt).tocoo()
+    # return sparse_to_tuple(adj_normalized)
+    return sparse_mx_to_torch_sparse_tensor(adj_normalized)
```

Comment:
I think there should be a pytorch-generic solution to this. Which part of these can't be achieved by `torch.sparse`? @dnguyen1196

Reply:
This file is also oudated. But true, I'll remove a lot of the unused functions in this file